src/_guides/language/coming-from/js-to-dart.md

```diff
@@ -0,0 +1,2806 @@
+  such as function parameters or list items.
+  This causes the formatter to place each item
+  onto its own line, which helps with readability,
+  especially when you have a lot of nested code
+  (as can happen in Flutter layout code).
+{{site.alert.end}}
+
+For more information on using commas to make your code
+read more like HTML, check out
+[Using trailing commas][] on flutter.dev.
+
+For more information about Dart conventions and linting,
+check out [Effective Dart][] and [Linter rules][].
+
+
+[Customizing static analysis]: /guides/language/analysis-options
+[`dart fix`]: /tools/dart-fix
+[Effective Dart]: /guides/language/effective-dart 
+[Linter rules]: /tools/linter-rules
+[Prettier]: https://prettier.io/
+[Using trailing commas]: {{site.flutter}}/development/tools/formatting#using-trailing-commas
+
+
+## Built-in types
+
+While both JavaScript and Dart have types,
+only Dart is a strongly typed language.
+This means that all Dart types have to be
+inferrable by the analyzer, explicitly defined,
+or assigned as `dynamic`, which disables static type
+checking for that identifier.
+
+Dart supports nullable and non-nullable versions
+of the the following built-in types:
+
+* Numbers (`int`, `double`)
+* Strings (String`)
+* Booleans (bool`)
+* Lists (`List`, also known as arrays)
+* Sets (`Set`)
+* Maps (`Map`)
+* Symbols (`Symbols`)
+* The value `null` (`Null`)
+
+For more information, check out
+[Built-in types][] in the [Dart Language Tour][].
+
+In Dart, all types are reference types,
+meaning that all variables refer to an object
+(an instance of a class). However, the types
+`int`, `double`, `String`, and `bool`
+are implemented to be immutable (unchanging)
+and are canonicalized,
+which means they behave as if they are value types.
+
+{{site.alert.note}}
+  JavaScript has two equality operators,
+  `==` and `===`. The `==` operator performs
+  the equality test after doing any necessary
+  type conversions. The `===` operator doesn't
+  perform type conversions. Dart doesn't have
+  an equivalent to `===`.
+{{site.alert.end}}
+
+For more information on these JavaScript operators,
+check out [this question on Stack Overflow][].
+
+[Built-in types]: /guides/language/language-tour#built-in-types
+[Dart Language Tour]: /guides/language
+[this question on Stack Overflow]: https://stackoverflow.com/questions/359494/which-equals-operator-vs-should-be-used-in-javascript-comparisons/359509#359509
+
+For example, the equals operator `==` and the `identical()`
+method are guaranteed to return true for the
+same values of these types, as shown in the following code:
+
+<!--skip-->
+Dart
+```dart
+var a = 1;
+var b = 1;
+
+print(a == b); // Prints true
+print(identical(a, b)); // Prints true
+```
+
+The next section covers basic types.
+Other types are covered later in this page,
+including collection types (lists, maps)
+and types that aid asynchrony (futures, streams).
+
+### Numbers
+
+Dart has three data types for holding numbers:
+
+`num`
```

Comment:
Should this start with ### as it's kinda acting as a header?

Reply:
Not a header, but thanks for pointing out the malformed html! fixed